### ISSUE TITLE
[Bug]: Docker container fails to start: `require': cannot load such file -- webrick (LoadError)

### ISSUE THREAD

**Issue by kpcyrd** [jekyll/jekyll#9066]
### Operating System

Docker

### Ruby Version

ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-linux-musl]

### Jekyll Version

jekyll 4.2.2

### Expected Behavior

jekyll should start without crashing

### Current Behavior

jekyll crashes

### Relevant log output

```shell
% docker run --rm --volume="$PWD:/srv/jekyll" -p 127.0.0.1:4000:4000 -p 127.0.0.1:35729:35729 -it jekyll/jekyll jekyll serve --trace
ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-linux-musl]
Configuration file: /srv/jekyll/_config.yml
            Source: /srv/jekyll
       Destination: /srv/jekyll/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
                    done in 1.212 seconds.
 Auto-regeneration: enabled for '/srv/jekyll'
<internal:/usr/local/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- webrick (LoadError)
	from <internal:/usr/local/lib/ruby/site_ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/commands/serve/servlet.rb:3:in `<top (required)>'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/commands/serve.rb:179:in `require_relative'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/commands/serve.rb:179:in `setup'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/commands/serve.rb:100:in `process'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `each'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
	from /usr/gem/gems/jekyll-4.2.2/lib/jekyll/commands/serve.rb:86:in `block (2 levels) in init_with_program'
	from /usr/gem/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
	from /usr/gem/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
	from /usr/gem/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
	from /usr/gem/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
	from /usr/gem/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
	from /usr/gem/gems/jekyll-4.2.2/exe/jekyll:15:in `<top (required)>'
	from /usr/gem/bin/jekyll:25:in `load'
	from /usr/gem/bin/jekyll:25:in `<main>'
%
```

**Comment by pranabdas** [comment]
Do you have `webrick` installed in your project? Could be same as https://github.com/github/pages-gem/issues/752

**Comment by iuriguilherme** [comment]
It is the same issue. OP have Ruby version above 3.0.0 and Jekyll below 4.3.0, therefore need to `gem install webrick` in the container's Ruby environment.

**Comment by caglareker** [comment]
> It is the same issue. OP have Ruby version above 3.0.0 and Jekyll below 4.3.0, therefore need to `gem install webrick` in the container's Ruby environment.

It worked for me. I added `gem "webrick"` to the Gemfile.

**Comment by kpcyrd** [comment]
Pinning to 4.2.0 works for me too.

Are there examples how the Gemfile is supposed to be used with the docker container? Jekyll 4.3.0 doesn't seem to exist yet.

**Comment by softwaregravy** [comment]
I am running this locally with ruby 3.1. With nothing specified (`gem "jekyll"`) I get failed to start. I'm guessing everyone trying to start anything new will hit this. 

It seems like jekyll has a hard dependency on webrick when running locally and the gemspec should be updated
_or_
Everywhere the `jekyll serve` command is mentioned ([ex](https://jekyllrb.com/docs/usage/)) it should say go install webrick 

As it is, following instructions in Jekyll's docs or Github pages' leads to failures.

**Comment by ashmaroli** [comment]
@softwaregravy This is a known issue and has been addressed in the `master` branch. The next feature release version (`v4.3.0`) will have `webrick` listed as a dependency and will load the library irrespective of the Ruby version.

**Comment by ashmaroli** [comment]
Considered as resolved with release of v4.3.0

**Comment by kpcyrd** [comment]
I think there's a problem with the docker image, `jekyll/jekyll:latest` and `jekyll/jekyll:4` both still contain `jekyll 4.2.2`, there's no tag for `4.3.0` or `4.3.1` either at https://hub.docker.com/r/jekyll/jekyll/tags.

**Comment by robjmills** [comment]
as above, looks like we're waiting on @envygeeks to tag those images?

**Comment by iuriguilherme** [comment]
> Considered as resolved with release of v4.3.0

I'm gonna repeat myself:

1. It's considered as resolved if it's fixed in every version of Jekyll that's affected by the issue and being used right now. 4.3 is not the only version being used, therefore it's not the only version needing to be fixed
2. How and where it has been released? There are multiple distribution channels and therefore many users still affected by this issue

**Comment by kpcyrd** [comment]
I think at this point it's more about "docker.io/jekyll/jekyll seems unmaintained and bugfixes don't land there anymore". The latest version there is `4.2.2` which was released almost a year ago at this point.

Would you be interested in a PR that adds github actions to automatically push releases to ghcr? The images would then show up over here:

![image](https://user-images.githubusercontent.com/7763184/217504608-cc6ad40b-2cf8-44c6-8814-7e19ecf7ab84.png)
